preference-service/src/controllers/preference.controller.ts

```diff
@@ -0,0 +1,95 @@
+import { Request, Response } from "express";
+import { PreferenceService } from "../services/preference.service";
+import { HttpStatus } from "../utils/httpStatus";
+import { ServerError, ErrorCode } from "../utils/serverError";
+import Messages from "../utils/Messages";
+
+export class PreferenceController {
+    public static async createPreference(req: Request, res: Response) {
+        const { userId, placeCategoryId, checkoutSlotId } = req.body;
+        if (!userId || !placeCategoryId || !checkoutSlotId) {
+            res.status(HttpStatus.BAD_REQUEST).json(new ServerError(
+                Messages.validation.preference_data_needs_to_be_provided,
+                ErrorCode.NOT_ENOUGH_DATA,
+            ).toJSON());
+
+            return;
+        }
+
+        if (isNaN(userId)) {
```

Comment:
```typescript
[userId, placeCategoryId, checkoutSlotId].forEach(param => {
    if (isNaN(param)) {
        res.status(HttpStatus.BAD_REQUEST).json(new ServerError(
            Messages.validation.id_must_be_number(param),
            ErrorCode.NOT_ENOUGH_DATA,
        ).toJSON());
    }
});
```

Reply:
Done

Reply:
@aacgn Now I see a problem with my approach. This is better:
```typescript
for (let param of [userId, placeCategoryId, checkoutSlotId])
    if (isNaN(param)) {
        res.status(HttpStatus.BAD_REQUEST).json(new ServerError(
            Messages.validation.id_must_be_number(param),
            ErrorCode.NOT_ENOUGH_DATA,
        ).toJSON());
        return;
    }
};
```